src/index.js

```diff
@@ -41,44 +50,50 @@ module.exports = function(babel) {
     }
   });
 
-  let hasDSImport = false;
   let hasAnyDataImport = false;
+  let decl = null;
+  let declPath;
 
   return {
     name: 'ember-data-packages-polyfill',
     visitor: {
       Program: {
-        enter() {
-          hasDSImport = false;
+        enter(path) {
           hasAnyDataImport = false;
+          decl = t.ImportDeclaration(
+            [t.ImportDefaultSpecifier(t.identifier('DS'))],
+            t.stringLiteral('ember-data')
+          );
+          path.unshiftContainer(
+            'body',
+            decl
+          );
         },
-        exit(path) {
-          if (hasAnyDataImport && !hasDSImport) {
-            // add `import DS from 'ember-data';`
-            path.unshiftContainer(
-              'body',
-              t.ImportDeclaration(
-                [t.ImportDefaultSpecifier(t.identifier('DS'))],
-                t.stringLiteral('ember-data')
-              )
-            );
+        exit() {
+          if (!hasAnyDataImport) {
+            // remove unnecessary `import DS from 'ember-data';`
+            declPath.remove();
           }
         },
       },
       ImportDeclaration(path, state) {
         let disallowedList = (state.opts && state.opts.disallowedList) || [];
         let node = path.node;
-        let replacements = [];
         let declarations = [];
         let removals = [];
         let specifiers = path.get('specifiers');
         let importPath = node.source.value;
 
+        if (path.node === decl) {
+          declPath = path;
+          return;
+        }
+
         if (importPath === 'ember-data') {
           // For `import DS from 'ember-data'`, we do nothing
           let specifierPath = specifiers.find(specifierPath => {
             if (specifierPath.isImportDefaultSpecifier()) {
```

Comment:
I think we actually have to ensure that it's name is `DS`, right? Like, there is nothing _wrong_ with:

```js
import EmberData from 'ember-data';
```

But our transform would basically break (because we'd remove the `import DS from 'ember-data'` that is added in `Program.enter` and modify to use `DS.*` but that wouldn't actually work)

Reply:
@rwjblue I covered this scenario in the AMD test case here: 

```ts
import EmberData from 'ember-data';
export var User = EmberData.Model;
```

What happens is we change this to:

```ts
import DS from 'ember-data';
var EmberData = DS;
export var User = EmberData.Model;
```

Reply:
Awesome, thanks for confirming with tests 😸